File: back/src/entity/raclotto_session.py
from datetime import datetime

from sqlalchemy import Column, Integer, String, DateTime, Boolean

from back.src.driver.database import BaseModel
from back.src.entity.mixin import SerializableMixin


class RaclottoSession(SerializableMixin, BaseModel):
    __tablename__ = "session"

    id = Column(Integer, primary_key=True)
    key = Column(String, nullable=False, unique=True)
    name = Column(String, nullable=False)
    timestamp = Column(DateTime, nullable=False, default=datetime.now())
    active = Column(Boolean, nullable=False, default=True)

File: front/src/model/raclottoSession.ts
export class RaclottoSession {
    public id: number;
    public key: string;
    public name: string;
    public timestamp: Date;
    public active: boolean;

    constructor() {
        this.id = 0;
        this.key = "";
        this.name = "";
        this.timestamp = new Date();
        this.active = false;
    }

    static fromParsed(parsed: { id: number; key: string; name: string; timestamp: Date | string; active: boolean }) {
        let session = new RaclottoSession();
        session.id = parsed.id;
        session.key = parsed.key;
        session.name = parsed.name;
        session.timestamp = parsed.timestamp instanceof Date ? parsed.timestamp : new Date(parsed.timestamp);
        session.active = parsed.active;
        return session;
    }
}